dialogue-core/src/main/java/com/palantir/dialogue/core/PinUntilErrorNodeSelectionStrategyChannel.java

```diff
@@ -182,6 +191,19 @@ private OptionalInt incrementHostIfNecessary(int pin) {
         @Value.Auxiliary
         int stableIndex();
 
+        @Value.Auxiliary
+        Ticker clock();
+
+        @Value.Derived
+        default CoarseExponentialDecayReservoir successReservoir() {
+            return new CoarseExponentialDecayReservoir(clock()::read, Duration.ofSeconds(10));
+        }
+
+        @Value.Derived
+        default CoarseExponentialDecayReservoir failureReservoir() {
+            return new CoarseExponentialDecayReservoir(clock()::read, Duration.ofSeconds(10));
+        }
```

Comment:
think this stuff was a intended for a separate PR right?

Reply:
doh